### PR TITLE
Add image for fast tests run (with clone, build, etc.)

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add docker image for fast tests.

### CODE REVIEW

File: docker/test/fasttest/run.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ln -s /usr/share/clickhouse-test/config/disks.xml /etc/clickhouse-server/config.d/
+ln -s /usr/share/clickhouse-test/config/secure_ports.xml /etc/clickhouse-server/config.d/
+ln -s /usr/share/clickhouse-test/config/clusters.xml /etc/clickhouse-server/config.d/
+ln -s /usr/share/clickhouse-test/config/graphite.xml /etc/clickhouse-server/config.d/
+ln -s /usr/share/clickhouse-test/config/server.key /etc/clickhouse-server/
+ln -s /usr/share/clickhouse-test/config/server.crt /etc/clickhouse-server/
+ln -s /usr/share/clickhouse-test/config/dhparam.pem /etc/clickhouse-server/
+ln -sf /usr/share/clickhouse-test/config/client_config.xml /etc/clickhouse-client/config.xml
+
+service zookeeper start && sleep 5
```
Comment (alexey-milovidov) [inline docker/test/fasttest/run.sh:67]: Do we need it?

Comment (alexey-milovidov) [reply docker/test/fasttest/run.sh:67]: The same question for usual functional tests.

Comment (alesapin) [reply docker/test/fasttest/run.sh:67]: No, I think we can remove zookeeper because we have testkeeper.

**Comment by alesapin** [comment]
Why it's so slow? Why tests failed?

File: docker/test/fasttest/run.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    if [ "$COMMIT_SHA" != "" ]; then
+        git checkout $COMMIT_SHA
+    fi
+fi
+
+
+
+git submodule update --init --recursive | ts '%Y-%m-%d %H:%M:%S' | tee /test_output/submodule_log.txt
+
+CMAKE_LIBS_CONFIG="-DENABLE_RDKAFKA=0 -DENABLE_S3=0 -DUSE_SENTRY=0 -DENABLE_AMQPCPP=0 -DENABLE_HDFS=0 -DENABLE_MYSQL=0 -DENABLE_GRPC=0 -DENABLE_CURL=0 -DENABLE_EMBEDDED_COMPILER=0"
```
Comment (alexey-milovidov) [inline docker/test/fasttest/run.sh:31]: Also check this list:
```
-D ENABLE_JEMALLOC=0 -D ENABLE_CAPNP=0 -D ENABLE_RDKAFKA=0 -D ENABLE_UNWIND=0 -D ENABLE_ICU=0 -D ENABLE_POCO_MONGODB=0 -D ENABLE_POCO_REDIS=0 -D ENABLE_POCO_NETSSL=0 -D ENABLE_ODBC=0 -D ENABLE_MYSQL=0 -D ENABLE_SSL=0 -D ENABLE_POCO_NETSSL=0 -D ENABLE_CASSANDRA=0
```

Comment (alexey-milovidov) [reply docker/test/fasttest/run.sh:31]: Also we can avoid cloning of most of submodules.

**Comment by alesapin** [comment]
Screenshot_2020-07-10 Task #720212633 CLICKHOUSE_FAST_TEST
Not so fast :(

**Comment by alesapin** [comment]
slitghly better
Screenshot_2020-07-10 Task #720978191 CLICKHOUSE_FAST_TEST

**Comment by alesapin** [comment]
image
Better

**Comment by alesapin** [comment]
image

**Comment by alesapin** [comment]
image

**Comment by alesapin** [comment]
Ok, I think we can merge it now.

File: docker/test/fasttest/run.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    if [ "$COMMIT_SHA" != "" ]; then
+        git checkout $COMMIT_SHA
+    fi
+fi
+
+SUBMODULES_TO_UPDATE="contrib/boost contrib/zlib-ng contrib/libxml2 contrib/poco contrib/libunwind contrib/ryu contrib/fmtlib contrib/base64 contrib/cctz contrib/libcpuid contrib/replxx contrib/double-conversion contrib/libcxx contrib/libcxxabi contrib/libc-headers contrib/lz4 contrib/zstd contrib/fastops contrib/rapidjson contrib/re2 contrib/sparsehash-c11"
+
+git submodule update --init --recursive $SUBMODULES_TO_UPDATE | ts '%Y-%m-%d %H:%M:%S' | tee /test_output/submodule_log.txt
+
+export CMAKE_LIBS_CONFIG="-DENABLE_RDKAFKA=0 -DENABLE_S3=0 -DUSE_SENTRY=0 -DENABLE_AMQPCPP=0 -DENABLE_HDFS=0 -DENABLE_MYSQL=0 -DENABLE_GRPC=0 -DENABLE_CURL=0 -DENABLE_JEMALLOC=0 -DENABLE_CAPNP=0 -DENABLE_RDKAFKA=0 -DENABLE_ICU=0 -DENABLE_POCO_NETSSL=0 -DENABLE_ODBC=0 -DENABLE_MYSQL=0 -DENABLE_SSL=0 -DENABLE_POCO_NETSSL=0 -DENABLE_CASSANDRA=0 -DENABLE_PARQUET=0 -DENABLE_AVRO=0 -DENABLE_EMBEDDED_COMPILER=0 -DENABLE_PROTOBUF=0 -DENABLE_H3=0 -DENABLE_TESTS=0 -DENABLE_UTILS=0 -DENABLE_ODBC=0 -DENABLE_GSASL_LIBRARY=0 -DENABLE_THINLTO=0 -DENABLE_MSGPACK=0 -DENABLE_LDAP=0 -DUSE_SNAPPY=0 -DENABLE_HYPERSCAN=0"
```
Comment (akuzm) [inline docker/test/fasttest/run.sh:31]: Maybe an allow-list with `ENABLE_LIBRARIES=0` and `ENABLE_MY_FAVORITE_LIBRARY=1`? This is going to break when we add new ones.

**Review by akuzm — APPROVED** [review]

**Comment by alesapin** [comment]
Merge after some stateless tests will pass.

**Comment by alesapin** [comment]
`00955_test_final_mark` not changed in this PR. Flap?

**Pull request merged**